### DISCUSSION TITLE
(How to?) ChatGPT with Vector Database and Conversation History

### DISCUSSION THREAD

**Discussion by neosandwich** [FlowiseAI/Flowise#418] [Q&A]
I'm specifically looking to have ChatGPT to consider the content of the documents I've uploaded to Pinecone. Moreover, I want the model to retain the memory of the previous four conversational exchanges. Presently, I'm experiencing difficulty in ensuring that the uploaded document is being factored into ChatGPT's responses. Thank you!

![image](https://github.com/FlowiseAI/Flowise/assets/112768024/5c338116-6cdd-421e-a824-99ea60a55517)

**Comment by weknock** [comment]
I'm attempting to do something similar, but each time I attempt to access the vector database, an error message pops up: "Chain conversational_retrieval_chain expects multiple inputs, cannot use 'run'." I'm unsure of what I'm doing incorrectly.
![image](https://github.com/FlowiseAI/Flowise/assets/71228954/66c3b60d-bde6-448e-9620-7fa5425b0c01)

**Reply by HenryHengZJ** [reply]
you cant use ConversationalRetrievalQAChain -> ChainTool, try using RetrievalQAChain

**Comment by HenryHengZJ** [comment]
> I'm specifically looking to have ChatGPT to consider the content of the documents I've uploaded to Pinecone. Moreover, I want the model to retain the memory of the previous four conversational exchanges. Presently, I'm experiencing difficulty in ensuring that the uploaded document is being factored into ChatGPT's responses. Thank you!
> 
> ![image](https://user-images.githubusercontent.com/112768024/248741670-5c338116-6cdd-421e-a824-99ea60a55517.png)

you'll have to use system prompt to tell chatgpt only return answer based on the document, refer to the system message placeholder
